src/Translation/Translation.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Astrotomic\Translation;
+
+use Illuminate\Support\Str;
+
+/**
+ * @property-read string $translatableModel
+ * @property-read string $translatableForeignKey
+ *
+ * @mixin Model
+ */
+trait Translation
+{
+    public function getTranslatableModelName(): string
+    {
+        return $this->translatableModel ?: $this->getTranslatableModelNameDefault();
+    }
+
+    public function getTranslatableModelNameDefault(): string
+    {
+        $modelName = get_class($this);
+
+        if ($namespace = $this->getTranslatableModelNamespace()) {
+            $modelName = $namespace.'\\'.class_basename(get_class($this));
+        }
+
+        return Str::replaceLast(config('translatable.translation_suffix', 'Translation'), '', $modelName);
+    }
+
+    public function getTranslatableModelNamespace(): ?string
+    {
+        return config('translatable.translatable_model_namespace');
+    }
+
+    public function getTranslatableRelationKey(): string
+    {
+        if ($this->translatableForeignKey) {
+            return $this->translatableForeignKey;
+        }
+
+        return Str::replaceFirst(Str::lower(config('translatable.translation_suffix', 'Translation')).'_', '', Str::snake(class_basename($this)).'_'.$this->getKeyName());
```

Comment:
No need for all this - the user can define the key otherwise we fallback to default Laravel behavior instead of introducing our own logic.

```suggestion
        return $this->translatableForeignKey;
```

Reply:
Thank's for the comment.
I get this error if I'm not define the key
```php
getTranslatableRelationKey() must be of the type string, null returned
```

What should I do?

Reply:
Ah, sorry - add a `?? null` to the line. The null coalescing operator will catch undefined property as well.

Reply:
Still not working. method returns `null`. where is the fallback to default Laravel behavior section? I don't understand

Reply:
It's where the method is used https://github.com/Astrotomic/laravel-translatable/pull/175/files#diff-a3b0f5f4d80d7bb75dbff6f3f8e609f3R47 - only the first argument is required. All others are optional/nullable, as soon as you pass `null` in it will use the default Laravel/ELoquent logic.

```php
// If no foreign key was supplied, we can use a backtrace to guess the proper
// foreign key name by using the name of the relationship function, which
// when combined with an "_id" should conventionally match the columns.
if (is_null($foreignKey)) {
    $foreignKey = Str::snake($relation).'_'.$instance->getKeyName();
}
```

Reply:
```php
// If no foreign key was supplied, we can use a backtrace to guess the proper
// foreign key name by using the name of the relationship function, which
// when combined with an "_id" should conventionally match the columns.
if (is_null($foreignKey)) {
    $foreignKey = Str::snake($relation).'_'.$instance->getKeyName();
}
```
Well, this part always returns `translatable_id` as `$foreignKey`. and if we want for example get the `Brand` model from `BrandTranslation` model, the foreign key should be `brand_id`. so there is no `translatable_id` in brands table. that's why I added that logic